src/components/application_manager/src/commands/mobile/subscribe_vehicle_data_request.cc

```diff
@@ -293,6 +293,20 @@ void SubscribeVehicleDataRequest::AddAlreadySubscribedVI(
   }
 }
 
+void SubscribeVehicleDataRequest::AddSpecificInfoToResponse(
+    smart_objects::SmartObject& response) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  AddDisallowedParameters(response);
+  if (helpers::Compare<mobile_apis::Result::eType, helpers::EQ, helpers::ONE>(
```

Comment:
@AKalinich-Luxoft 
As far as I see the only logic here is to decide whether info string needs to be cleared. So you can 
- move AddSpecificInfoToResponse, AddDisallowedParametersToInfo to private section of CommandRequestImpl
- create e.g. 'virtual bool DoCleanInfo(result_code)' in CommandRequestImpl class in private section and override it in commands subclasses
- call DoCleanInfo from AddSpecificInfoToResponse in CommandRequestImpl

Also I think it should be possible also to move to private Add/RemoveDisallowedParameters interfaces.

Reply:
@dev-gh clearing info is not only logic in overriden functions. `AddDisallowedParameters()` and `AddDisallowedParametersToInfo()` also should be called in these overrides. These functions are used only here. So creating separate `DoCleanInfo()` virtual function is not a good idea. This is a reason why I added `AddSpecificInfoToResponse()` function.
As alternative I can move functions like Add/RemoveDisallowedParameters, AddDisallowedParametersToInfo for example to `MessageHelper` because this is not default behaviour for all classes inherited from `CommandRequestImpl`.

Reply:
@AKalinich-Luxoft 
Moving code is even better, so move as much as possible out of CommandRequestImpl. I think its better to create another "command helper" space for such utils used by commands since MessageHelper is overwhelmed already with other stuff.

Reply:
@dev-gh good idea. I added new helper for commands and moved part of functionality related to disallowed params to this helper. I believe that it is possible to move more functions from command base class, however it will be a lot of changes for PR with fix, so it will have a sense to do in a separate pull request.
Please review final result in 4652e3534f26ee37da3b56a5ebe8345c22266278